tests/acceptance/clean-test-slow.js

```diff
@@ -0,0 +1,157 @@
+    var bowerrcPath = path.join(root, 'tmp', appName, '.bowerrc');
+    var bowerrc = JSON.parse(fs.readFileSync(bowerrcPath));
+    var updated = _.merge(bowerrc, settings);
+
+    return fs.writeFileSync(bowerrcPath, JSON.stringify(updated));
+  }
+
+  function initApp(appName) {
+    return ember([
+      'new',
+      appName,
+      '--skip-npm',
+      '--skip-bower'
+    ])
+    .then(function() {
+        var bowerCachePath = path.join(root, 'tmp/bower-cache');
+        return updateBowerrc({
+          storage: {
+            packages : path.join(bowerCachePath, 'packages'),
+            registry : path.join(bowerCachePath, 'registry'),
+            links : path.join(bowerCachePath, 'links')
+          }
+        });
+    })
+    .then(function() {
+        var npmrcPath = path.join(root, 'tmp', appName, '.npmrc');
+        var npmCachePath = path.join(root, 'tmp/npm-cache');
+
+        return fs.writeFileSync(npmrcPath, 'cache = ' + npmCachePath);
+    });
+  }
+
+  function assertFileNotExists(file) {
+    var filePath = path.join(process.cwd(), file);
+    expect(!existsSync(filePath), 'expected ' + file + ' not to exist');
+  }
+
+  function assertFileExists(file) {
+    var filePath = path.join(process.cwd(), file);
+    expect(existsSync(filePath), 'expected ' + file + ' to exist');
+  }
+
+  beforeEach(function() {
+    return tmp.setup('./tmp')
+      // possibly `acceptance.createTestTargets()` should be used instead
+      .then(function() {
+        process.chdir('./tmp');
+        return initApp(appName);
+      })
+      .then(function() {
+        return tmp.setup(path.join('tmp', appName, 'tmp'));
+      }).then(function() {
+        return tmp.setup(path.join('tmp', appName, 'dist'));
+      }).then(function() {
+        return tmp.setup(path.join('tmp', appName, customBowerDir));
+      }).then(function() {
+        return tmp.setup(path.join('tmp', appName, 'bower_components'));
+      }).then(function() {
+        return tmp.setup(path.join('tmp', appName, 'node_modules'));
+      });
+  });
+
+  afterEach(function() {
+    process.chdir(root);
+    return tmp.teardown('./tmp');
+  });
+
+  it('all', function() {
+    process.chdir('./tmp/' + appName);
+
+    return ember(['clean']).then(function() {
+      assertFileNotExists('tmp');
+      assertFileNotExists('dist');
+      assertFileNotExists('bower_components');
+      assertFileNotExists('node_modules/');
+    });
+  });
+
+  it('skip npm', function() {
+    process.chdir('./tmp/' + appName);
+
+    return ember([
+      'clean',
+      '--skip-npm'
+    ]).then(function() {
+      assertFileNotExists('tmp');
+      assertFileNotExists('dist');
+      assertFileNotExists('bower_components');
+      assertFileExists('node_modules/');
+    });
+  });
+
+  // @todo: Fix cache settings .npmrc
+  // Currently looks like lib/utilities/npm doesn't take it in account
+  // which causes global cache clean
+  xit('skip bower', function() {
```

Comment:
Currently this test cleans global cache from `~/.npm`. Looks like npm task should pass npm config to `utilities/npm`.